rules/scala_proto/private/ScalaProtoWorker.scala

```diff
@@ -65,6 +66,7 @@ object ScalaProtoWorker extends WorkerMain[Unit] {
 
   protected def work(ctx: Unit, args: Array[String], out: PrintStream, workDir: Path): Unit = {
     val workRequest = ScalaProtoRequest(workDir, ArgsUtil.parseArgsOrFailSafe(args, argParser, out))
+    InterruptUtil.throwIfInterrupted()
```

Comment:
I'm not aware of a better way to do this. I agree that this is something people will likely forget to do.

Good news is that the failure mode is the task running to completion rather than something more nefarious.

There is a deprecated way of killing a thread because it is unsafe.

SIGINT will stop the process, at which point the OS deals with cleaning things up. I'm not aware of an equivalent in the JVM land.

Reply:
> SIGINT will stop the process, at which point the OS deals with cleaning things up. I'm not aware of an equivalent in the JVM land.
Ah, this didn't occur to me.

It's unfortunate there isn't a better way to do this.